control.lua

```diff
@@ -40,7 +46,7 @@ local function processRequests(req, requests)
     if (not (log_point and log_point.valid)) then
         return
     end
-    if (not (log_point.mode == defines.logistic_mode.buffer or log_point.mode == defines.logistic_mode.requester)) then
+    if (not ((log_point.mode == defines.logistic_mode.buffer and buffer_chests_enabled) or log_point.mode == defines.logistic_mode.requester)) then
```

Comment:
Could also just check the setting value here directly, thus eliminating the need for any other change to control.lua
The `global.settings` should always be up to date with the current game settings.
For readability sakes, would rather just have that local be defined local right above this line, rather than be a file wide variable.